src/mjolnir/graphfilter.cc

```diff
@@ -219,7 +393,359 @@ void FilterTiles(GraphReader& reader,
            std::to_string(n_original_nodes));
   LOG_INFO("Filtered " + std::to_string(n_filtered_edges) + " directededges out of " +
            std::to_string(n_original_edges));
-  LOG_INFO("Can aggregate: " + std::to_string(can_aggregate));
+  LOG_INFO("Nodes to aggregate: " + std::to_string(can_aggregate));
+}
+
+void GetAggregatedData(GraphReader& reader,
+                       std::list<PointLL>& shape,
+                       GraphId& en,
+                       const GraphId& from_node,
+                       const graph_tile_ptr& tile,
+                       const DirectedEdge* directededge) {
+
+  // Get the tile at the end node.  Skip if node is in another tile.
+  // mode_change is not set for end nodes that are in diff tiles
+  if (directededge->endnode().tile_value() == tile->header()->graphid().tile_value()) {
+
+    // original edge data.
+    auto edgeinfo = tile->edgeinfo(directededge);
+    const NodeInfo* en_info = tile->node(directededge->endnode().id());
+
+    // do we need to aggregate?
+    if (en_info->mode_change()) {
+      std::unordered_set<std::string> isos;
+      bool isForward = directededge->forward();
+      auto id = directededge->endnode();
+      if (!isForward) {
+        std::reverse(shape.begin(), shape.end());
+      }
+      // walk in the correct direction.
+      uint64_t wayid = edgeinfo.wayid();
+      if (Aggregate(id, reader, shape, en, from_node, wayid, isos, isForward)) {
+        aggregated++; // count the current edge
+        // flip the shape back for storing in edgeinfo
+        if (!isForward) {
+          std::reverse(shape.begin(), shape.end());
+        }
+      }
+    }
+  }
+}
+
+// If we cross into another country we can't aggregate the edges
+// as the access can be different in each country.  Also, bollards
+// and or gates could exist at the node blocking access.
+//
+// Also, we need to handle islands that we created by tossing the
+// pedestrian edges.  For example://
+// https://www.openstreetmap.org/way/993706522
+// https://www.openstreetmap.org/way/975845893
```

Comment:
nice crazy examples :D I guess with default reachability constraints loki wouldn't ever snap inside these islands would it?

Reply:
yeah these islands exist in the data now.  Not too many of them, but would love to toss just in case someone started there

Reply:
that would be my guess too